File: 2025/04 - APR/11Apr25.cpp
class Solution {
  public:
    vector<int> dijkstra(int V, vector<vector<int>> &edges, int src) {
        // Code here
        vector<int> dis(V, INT_MAX);
        dis[src] = 0;
        
        vector<vector<pair<int, int>>> adj(V);
        for(auto z : edges){
            adj[z[0]].push_back({z[1], z[2]});
            adj[z[1]].push_back({z[0], z[2]});
        }
        
        
        queue<pair<int, int>> q;
        q.push({src, 0});
        
        while(!q.empty()){
            auto ele = q.front();
            q.pop();
            
            for(auto x : adj[ele.first]){
                int w = x.second;
                if((dis[ele.first]+ w)< dis[x.first]){
                    dis[x.first] = dis[ele.first]+ w;
                    q.push({x.first, dis[x.first]});
                }
            }
            
        }
        
        return dis;
        
    }
};

File: 2025/04 - APR/11Apr25.py
from collections import defaultdict
class Solution:
    def dijkstra(self, V, edges, src):
        graph = defaultdict(list)
        for u, v, w in edges:
            graph[u].append((v, w))
            graph[v].append((u, w))
        dist = [float('inf')] * V
        dist[src] = 0
        visited = set()
        while len(visited) < V:
            u = -1
            min_dist = float('inf')
            for i in range(V):
                if i not in visited and dist[i] < min_dist:
                    min_dist = dist[i]
                    u = i
            if u == -1:
                break 
            visited.add(u)
            for v, weight in graph[u]:
                if v not in visited and dist[u] + weight < dist[v]:
                    dist[v] = dist[u] + weight
        return dist
